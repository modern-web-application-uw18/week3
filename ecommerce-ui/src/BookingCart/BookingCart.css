.Booking-Cart-container {
    display: flex;
    flex-wrap: wrap;
  }