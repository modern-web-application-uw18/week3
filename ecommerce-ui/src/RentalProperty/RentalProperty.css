.Rental {
    width: 600px;
    height: 400px;
    border-width: 1px;
    align-items: center;
    flex-basis: 32%;
    margin-top: 20px;
}
  
  .Rental-image {
    max-width: 300px;
    padding: 10px;    
  }

  .Rental-title {
    font-weight: bold;
    text-align: center;    
  }

  .Rental-button {
    margin: 2px;
  }
  