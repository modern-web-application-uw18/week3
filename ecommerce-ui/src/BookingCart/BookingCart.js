import React, { Component } from 'react';
import PropTypes from 'prop-types';
import RentalProperty from './../RentalProperty/RentalProperty';
import './BookingCart.css';


const propTypes = {
  rentalProperties: PropTypes.arrayOf(Object)
}

class BookingCart extends Component {

  constructor(props){
    super(props);
    this.state = {
      total:0
    }
    this.calculateTotal = this.calculateTotal.bind(this);
  }

  calculateTotal(price) {
    this.setState({
      total: this.state.total + price
    });
  }

  render() {
    return (
      <div>
        <div className="Booking-Cart-container">
          {this.props.rentalProperties.map((rentalProperty, key) => {
            return <RentalProperty key={key} rentalProperty={rentalProperty} handleTotal={this.calculateTotal}/>;
          })}
        </div>
        <h2>Total Cart Summary: {this.state.total}$</h2>
      </div>
    );
  }
}

BookingCart.propTypes = propTypes;

export default BookingCart;
