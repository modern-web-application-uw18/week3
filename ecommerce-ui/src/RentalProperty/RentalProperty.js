import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './RentalProperty.css';

const propTypes = {
    rental: PropTypes.object
}

class RentalProperty extends Component {
    constructor(props){
        super(props);
        this.state = {
            added: false,
            removed: true
        }
    }

    addToCart = () => {
        this.setState({
            added: true,
            removed:false
        });
        this.props.handleTotal(this.props.rentalProperty.payment.cost);
    }

    removeFromCart = () => {
        this.setState({
            added: false,
            removed: true
        });
        this.props.handleTotal(-this.props.rentalProperty.payment.cost);
    }

    renderRental() {
        return (
            <div className="Rental">
                <div>
                    <button style={this.state.removed ? {className:"Rental-button"} : {display: 'none'}}  onClick={this.addToCart}>Add to Cart</button>
                    <button style={this.state.added ? {className:"Rental-button"} : {display: 'none'}} onClick={this.removeFromCart}>Remove from Cart</button>
                </div>
                <img className="Rental-image" src={this.props.rentalProperty.image} alt={this.props.rentalProperty.title} />
                <div className="Rental-title">{this.props.rentalProperty.title}</div>
                <div>{this.props.rentalProperty.payment.cost}$ per night - {this.props.rentalProperty.payment.description}</div>
            </div>
        )
    }

    render() {
        return this.renderRental();
    }
}

RentalProperty.propTypes = propTypes;

export default RentalProperty;