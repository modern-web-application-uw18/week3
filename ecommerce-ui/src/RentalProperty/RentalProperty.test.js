import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import RentalProperty from './RentalProperty';
import rentalProperties from './../airbnbs.json';
import BookingCart from './../BookingCart/BookingCart'

let rentalProperty = rentalProperties[0];

describe('Rental property component', () => {
    it('should render', () => {
        const component = ReactTestRenderer.create(<RentalProperty rentalProperty={rentalProperty}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});