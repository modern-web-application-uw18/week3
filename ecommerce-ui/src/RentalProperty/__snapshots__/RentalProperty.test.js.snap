// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rental property component should render 1`] = `
<div
  className="Rental"
>
  <div>
    <button
      onClick={[Function]}
      style={
        Object {
          "className": "Rental-button",
        }
      }
    >
      Add to Cart
    </button>
    <button
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      Remove from Cart
    </button>
  </div>
  <img
    alt="Centrally-located Manhattan studio"
    className="Rental-image"
    src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
  />
  <div
    className="Rental-title"
  >
    Centrally-located Manhattan studio
  </div>
  <div>
    105
    $ per night - 
    Free cancellation
  </div>
</div>
`;
